spot_wrapper/wrapper.py

```diff
@@ -2047,25 +2056,56 @@ def _upload_graph_and_snapshots(self, upload_filepath):
             )
         for waypoint in self._current_graph.waypoints:
             # Load the waypoint snapshots from disk.
-            with open(
-                upload_filepath + "/waypoint_snapshots/{}".format(waypoint.snapshot_id),
-                "rb",
-            ) as snapshot_file:
+            if len(waypoint.snapshot_id) == 0:
+                continue
+            waypoint_filepath = os.path.join(
+                upload_filepath, "waypoint_snapshots", waypoint.snapshot_id
+            )
+            if not os.path.exists(waypoint_filepath):
+                continue
+            with open(waypoint_filepath, "rb") as snapshot_file:
                 waypoint_snapshot = map_pb2.WaypointSnapshot()
                 waypoint_snapshot.ParseFromString(snapshot_file.read())
                 self._current_waypoint_snapshots[
                     waypoint_snapshot.id
                 ] = waypoint_snapshot
+
+                for fiducial in waypoint_snapshot.objects:
+                    if not fiducial.HasField("apriltag_properties"):
+                        continue
+
+                    str_id = str(fiducial.apriltag_properties.tag_id)
+                    if (
+                        str_id in self._current_anchored_world_objects
+                        and len(self._current_anchored_world_objects[str_id]) == 1
+                    ):
+                        # Replace the placeholder tuple with a tuple of (wo, waypoint, fiducial).
+                        anchored_wo = self._current_anchored_world_objects[str_id][0]
+                        self._current_anchored_world_objects[str_id] = (
+                            anchored_wo,
+                            waypoint,
+                            fiducial,
+                        )
+
         for edge in self._current_graph.edges:
             # Load the edge snapshots from disk.
-            with open(
-                upload_filepath + "/edge_snapshots/{}".format(edge.snapshot_id), "rb"
-            ) as snapshot_file:
+            if len(edge.snapshot_id) == 0:
+                continue
+            edge_filepath = os.path.join(
+                upload_filepath, "edge_snapshots", edge.snapshot_id
+            )
+            if not os.path.exists(edge_filepath):
+                continue
+            with open(edge_filepath, "rb") as snapshot_file:
                 edge_snapshot = map_pb2.EdgeSnapshot()
                 edge_snapshot.ParseFromString(snapshot_file.read())
                 self._current_edge_snapshots[edge_snapshot.id] = edge_snapshot
+        for anchor in self._current_graph.anchoring.anchors:
+            self._current_anchors[anchor.id] = anchor
```

Comment:
I want to be sure that it is correct to mutate this (possibly leaving "state" anchor ids from the last time it ran) instead of creating `_current_anchors` from scratch with juts the new information.

Reply:
The code again comes from BD example. The functionality that uses current_anchors hasn't been added yet. It is useful for informing what the frame of reference is for the point cloud map.